Source/ROSIntegrationVision/Private/VisionActor.cpp

```diff
@@ -2,910 +2,30 @@
-	std::thread ThreadColor, ThreadDepth, ThreadObject;
-	bool DoColor, DoDepth, DoObject;
-	bool DoneColor, DoneObject;
-};
-
 // Sets default values
-AVisionActor::AVisionActor() : ACameraActor(), Width(960), Height(540), Framerate(1), FieldOfView(90.0), ServerPort(10000), FrameTime(1.0f / Framerate), TimePassed(0), ColorsUsed(0)
+AVisionActor::AVisionActor() : AActor()
 {
 	UE_LOG(LogTemp, Warning, TEXT("VisionActor CTOR"));
-	Priv = new PrivateData();
 
 	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
 	PrimaryActorTick.bCanEverTick = true;
-
-	UE_LOG(LogTemp, Warning, TEXT("Creating color camera."));
-	Color = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("ColorCapture"));
-	Color->SetupAttachment(RootComponent);
-	Color->CaptureSource = ESceneCaptureSource::SCS_FinalColorLDR;
-	Color->TextureTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("ColorTarget"));
-	Color->TextureTarget->InitAutoFormat(Width, Height);
-	Color->FOVAngle = FieldOfView;
-
-	UE_LOG(LogTemp, Warning, TEXT("Creating depth camera."))
-		Depth = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("DepthCapture"));
-	Depth->SetupAttachment(RootComponent);
-	Depth->CaptureSource = ESceneCaptureSource::SCS_FinalColorLDR;
-	Depth->TextureTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("DepthTarget"));
-	Depth->TextureTarget->InitAutoFormat(Width, Height);
-	Depth->FOVAngle = FieldOfView;
-
-	UE_LOG(LogTemp, Warning, TEXT("Creating object camera."))
-		Object = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("ObjectCapture"));
-	Object->SetupAttachment(RootComponent);
-	Object->CaptureSource = ESceneCaptureSource::SCS_FinalColorLDR;
-	Object->TextureTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("ObjectTarget"));
-	Object->TextureTarget->InitAutoFormat(Width, Height);
-	Object->FOVAngle = FieldOfView;
-
-	UE_LOG(LogTemp, Warning, TEXT("Loading materials"))
-		ConstructorHelpers::FObjectFinder<UMaterial> MaterialDepthFinder(TEXT("Material'/ROSIntegrationVision/SceneDepth.SceneDepth'"));
-	if (MaterialDepthFinder.Object != nullptr)
-	{
-		MaterialDepthInstance = UMaterialInstanceDynamic::Create(MaterialDepthFinder.Object, Depth);
-		if (MaterialDepthInstance != nullptr)
-		{
-			Depth->PostProcessSettings.AddBlendable(MaterialDepthInstance, 1);
-		}
-	}
-	else
-		UE_LOG(LogTemp, Error, TEXT("Could not load material for depth."));
-
-}
-
-AVisionActor::~AVisionActor()
-{
-	delete Priv;
-	//delete _camera_info_publisher;
-	//delete _image_publisher;
-	//delete _depth_publisher;
-	//delete _tf;
-	UE_LOG(LogTemp, Display, TEXT("VisionActor got destroyed"));
+    
+    RootComponent = CreateDefaultSubobject<USceneComponent>(TEXT("Root"));
+    SetRootComponent(RootComponent);
+    
+    vision = CreateDefaultSubobject<UVisionComponent>(TEXT("Vision"));
+    vision->ParentLink = "/world";
+    vision->SetupAttachment(RootComponent);

```

Comment:
This is the line where the `VisionComponent` is attached to the `VisionActor`'s `RootComponent`.